servant/src/Servant/API/Fragment.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE DataKinds          #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE PolyKinds          #-}
+{-# LANGUAGE TypeOperators      #-}
+{-# OPTIONS_HADDOCK not-home    #-}
+module Servant.API.Fragment (Fragment) where
+
+import           Data.Typeable
+                 (Typeable)
+
+-- | Lookup the value associated to the URI fragment
+-- and try to extract it as a value of type @a@.
+--
+-- Example:
+--
+-- >>> -- /post#TRACKING
+-- >>> type MyApi = "post" :> Fragment Text :> Get '[JSON] Tracking
+data Fragment (a :: *)
```

Comment:
`Fragment` behaves similarly to `QueryParam`, so I think we should add modifiers. I think that we should at least recognise `Lenient`/`Strict` and `Description` as we do for `QueryParam`. I am not sure if `Required` makes sense here, but we could also add `Required`/`Optional`.

Reply:
- Modifiers added.
- Server restricted to `FoldLenient`.
- Fragment itself set by default to `Lenient` to allow user control of parse errors handling.